### PR TITLE
Support new type in bundles

### PR DESCRIPTION
**Pull request by shiraji**
see #1577

### CODE REVIEW

**Comment by shiraji** [comment]
As we discuss in the chat room, we need to think more test cases for this pull request. (or comment on SaveInstanceStateActivityParameterizedTest)

File: AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/instancestate/SaveInstanceStateActivity.java [hunk 1]
```diff
@@ -183,4 +190,7 @@
 
 	@InstanceState
 	ParcelerBean parcelerBean;
+
+	@InstanceState
+	SparseArray<MyGenericParcelableBean<Integer>> mySparseParcelableArray;
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/instancestate/SaveInstanceStateActivity.java]: Let's add a comment for now.

Comment (shiraji) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/instancestate/SaveInstanceStateActivity.java]: Sure.

**Comment by shiraji** [comment]
I created compiling test cases for failing cases, since I don't believe my manual tests.

**Comment by WonderCsabo** [comment]
Great. But let's add a positive test-case and check the generated code is right, then.

**Comment by shiraji** [comment]
Cool. I will add them tomorrow.

**Comment by shiraji** [comment]
There you go.

File: AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/helper/BundleSparseArrayCompileSuccessActivity.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import android.app.Activity;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.util.SparseArray;
+
+@EActivity
+public class BundleSparseArrayCompileSuccessActivity extends Activity {
+
+	@InstanceState
+	SparseArray<Parcelable> mSparseArrayWithParcelable;
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/helper/BundleSparseArrayCompileSuccessActivity.java]: Please do not use this naming convention.

Comment (shiraji) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/helper/BundleSparseArrayCompileSuccessActivity.java]: Oops. I will fix that and another one.

**Comment by WonderCsabo** [comment]
Looks good. Can you squash the commits?

**Comment by shiraji** [comment]
Squashed.

File: AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/helper/BundleSparseArrayActivityTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	public void setUp() {
+		addManifestProcessorParameter(BundleSparseArrayActivityTest.class);
+		addProcessor(AndroidAnnotationProcessor.class);
+		addManifestProcessorParameter(BundleSparseArrayActivityTest.class, "AndroidManifest.xml");
+	}
+
+	@Test
+	public void compileSuccessForBundleSparseArray() {
+		CompileResult result = compileFiles(BundleSparseArrayCompileSuccessActivity.class);
+		assertCompilationSuccessful(result);
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/helper/BundleSparseArrayActivityTest.java]: Sorry, i think i was not clear. Please add `assertGeneratedClassContains()` here to actually check the correct method call was generated.

Comment (shiraji) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/helper/BundleSparseArrayActivityTest.java]: Sure

**Comment by WonderCsabo** [comment]
Nice. Please squash the last commit and this is can be merged.

**Comment by shiraji** [comment]
Squashed!

**Comment by WonderCsabo** [comment]
Great job, thanks!

**Comment by shiraji** [comment]
NP! Thanks for your help!

**Pull request merged**